lib/svgo/coa.js

```diff
@@ -300,23 +301,32 @@ function optimizeFolder(config, dir, output) {
  * @param {string} output output directory
  * @return {Promise}
  */
-function processDirectory(config, dir, files, output) {
+async function processDirectory(config, dir, files, output) {
   // take only *.svg files, recursively if necessary
   var svgFilesDescriptions = getFilesDescriptions(config, dir, files, output);
-
-  return svgFilesDescriptions.length
-    ? Promise.all(
-        svgFilesDescriptions.map((fileDescription) =>
-          optimizeFile(
-            config,
-            fileDescription.inputPath,
-            fileDescription.outputPath,
-          ),
-        ),
-      )
-    : Promise.reject(
-        new Error(`No SVG files have been found in '${dir}' directory.`),
-      );
+  
+  if ( svgFilesDescriptions.length == 0 ) {
+    throw 'No SVG files have been found in "' + dir + '" directory.';
+  }
+  
+  let results = [ ],
+      cpus = os.cpus().length;
+
+  while ( svgFilesDescriptions.length > 0 ) {
+    let _svgFilesDescriptions = svgFilesDescriptions.splice(0, cpus);
+    
+    _svgFilesDescriptions = _svgFilesDescriptions.map(function(fileDescription) {
```

Comment:
I'm not sure if this line errors or not, I would expect it to since with TS once a variable has a type you don't change it. Usually here you would create a new variable to store these results.

Reply:
Maybe you are right, i dont use TS, i do this changes over the npm downloaded code